src/Components/FashCard/FlashCard.tsx

```diff
@@ -1,23 +1,22 @@
-import { useContext, useState } from "react";
+import { useContext } from "react";
 import { Card, Button } from "react-bootstrap";
 import { MyQuestionContext } from "../QuestionContext/QuestionContext";
 import { useParams } from "react-router-dom";
-import { Question, getQuestions } from "../../Apis/types";
+import { getQuestions } from "../../Apis/types";
 import Timer from "../Timer/Timer";
 import { Loading } from "../Loading/Loading";
 import { ArrayMethods } from "../ArrayQuestions/ArrayMethods";
 
 export const FlashCard = () => {
     const { questions } = useContext(MyQuestionContext);
     const { questionId } = useParams();
-    const randomId = Math.floor(
-        Math.random() * questions.length + 1
-    ).toString();
-    const [randomQuestion, setRandomQuestion] = useState<Question | undefined>(
-        getQuestions(questions, questionId)
-    );
+    const randomId = () =>
+        Math.floor(Math.random() * questions.length).toString();
+    let randomQuestion = getQuestions(questions, questionId);
```

Comment:
was changed from state to variable b/c the getQuestions was not done before the setState so it was being set to `undefined`. decided to use a var instead of a useEffect so there was less code and felt cleaner.